docs/torch/node_class.rst

```diff
@@ -34,261 +44,221 @@ DeepGNN models typically contain multiple parts:
+    >>> def train_func(config: Dict):
+    ...     # Set random seed
+    ...     train.torch.enable_reproducibility(seed=0)
+    ...
+    ...     # Initialize the model and wrap it with Ray
+    ...     model = GAT(in_dim=1433, num_classes=7)
+    ...     model = train.torch.prepare_model(model)
+    ...
+    ...     # Initialize the optimizer and wrap it with Ray
+    ...     optimizer = torch.optim.Adam(model.parameters(), lr=.005, weight_decay=0.0005)
+    ...     optimizer = train.torch.prepare_optimizer(optimizer)
+    ...
+    ...     # Define the loss function
+    ...     loss_fn = nn.CrossEntropyLoss()
+    ...
+    ...     # Dataset
+    ...     g = Client("/tmp/cora", [0])
+    ...     q = GATQuery()
+    ...     dataset = TorchDeepGNNDataset(
+    ...         sampler_class=FileNodeSampler,
+    ...         backend=g,
+    ...         query_fn=q.query,
+    ...         prefetch_queue_size=2,
+    ...         prefetch_worker_size=2,
+    ...         sample_files="/tmp/cora/train.nodes",
+    ...         batch_size=140,
+    ...         shuffle=True,
+    ...         drop_last=True,
+    ...         worker_index=0,
+    ...         num_workers=1,
+    ...     )
+    ...     dataset = torch.utils.data.DataLoader(
+    ...         dataset=dataset,
+    ...         num_workers=0,
+    ...     )
+    ...
+    ...     # Execute the training loop
+    ...     model.train()
+    ...     for epoch in range(1):
+    ...         for i, batch in enumerate(dataset):
+    ...             scores = model(batch)
+    ...             labels = batch["labels"][batch["input_mask"]].flatten()
+    ...             loss = loss_fn(scores.type(torch.float32), labels)
+    ...             optimizer.zero_grad()
+    ...             loss.backward()
+    ...             optimizer.step()
+    ...
+    ...             session.report({"metric": (scores.argmax(1) == labels).sum(), "loss": loss.item()})
+
+In this step we start the training job.
+First we start a local ray cluster with `ray.init() <https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-init>`.
+Next we initialize a `TorchTrainer <https://docs.ray.io/en/latest/ray-air/package-ref.html#pytorch>`
+object to wrap our training loop. This takes parameters that go to the training loop and parameters
+to define number workers and cpus/gpus used.
+Finally we call trainer.fit() to execute the training loop.
 
 .. code-block:: python
 
-	>>> run_dist(
-	...     init_model_fn=create_model,
-	...     init_dataset_fn=create_dataset,
-	...     init_optimizer_fn=create_optimizer,
-	...     init_args_fn=init_args,
-	...		run_args=arg_list,
-	... )
+    >>> ray.init()
+    RayContext(...)
+    >>> trainer = TorchTrainer(
+    ...     train_func,
+    ...     train_loop_config={},
+    ...     run_config=RunConfig(verbose=0),
+    ...     scaling_config=ScalingConfig(num_workers=1, use_gpu=False),
+    ... )
+    >>> result = trainer.fit()
```

Comment:
what is the expected outcome? Are we going to evaluate our model and check accuracy?

Reply:
I've added eval step.